### ISSUE TITLE
Not able to add file to remote server

### ISSUE THREAD

**Issue by prav02** [chunkai1312/multer-sftp#5]
When I ran the sample code give in the example with my server credentials and try to add a file in remote server. Instead of add the file in the remote server the code add the file on my local.  Do I need to do any other modification to add the file in remote server??

**Comment by chunkai1312** [comment]
Hi @prav02, sorry for late reply. Please change `host` to your remote server's, then try it again.

**Comment by prav02** [comment]
@chunkai1312  Yes, I already tried that, but still it store file in local system instead of my remote server.

**Comment by chunkai1312** [comment]
@prav02 I tried my code just now and confirmed that it works fine with me. If it not solved then post your sample code here. Thanks.

**Comment by prav02** [comment]
// Multer For Remote Server
try{
var storage = sftpStorage({
    sftp: {
      host: 'My remote server Ip here',
      port: My remote server port here,
      username: 'remote server username here',
      password: 'password here'
    },
    destination: function (req, file, cb)  {
        var foldername = 'public/uploads/Piping/demo/'
        if (fs.existsSync(foldername)) {
            // fs.mkdirSync('public/uploads/project/demo2')
            cb(null, 'public/uploads/Piping/demo/')
            console.log('File Successfully upload on Remote Server')
        }else{
            console.log('The folder Does not exist')
        }
    },
    filename: function (req, file, cb) {
        console.log('FIle Name:'+JSON.stringify (file.originalname + '-' + Date.now()))
      cb(null, file.originalname + '-' + Date.now())
    }
  })
} catch (e){
    console.log(e)
}

  var upload = multer({ storage: storage })

**Comment by chunkai1312** [comment]
[Multer](https://github.com/expressjs/multer) is middleware for handling `multipart/form-data`, so you should setup your express server.
The following is my complete example:

```js
const express = require('express')
const multer = require('multer')
const sftpStorage = require('multer-sftp')

const app = express()

const storage = sftpStorage({
  sftp: {
    host: 'your-remote-server-host',
    port: 22,
    username: 'your-username',
    password: 'your-password'
  },
  destination: function (req, file, cb) {
    cb(null, '/path/to/upload')
  },
  filename: function (req, file, cb) {
    cb(null, file.fieldname + '-' + Date.now())
  }
})

const upload = multer({ storage: storage })

app.post('/upload', upload.single('file'), (req, res, next) => {
  res.send('ok')
})

app.listen(3000, () => {
  console.log('express server listening on port 3000')
})
```

If you want to upload files via ssh directly, you can use [ssh2-sftp-client](https://github.com/theophilusx/ssh2-sftp-client) instead of this module.

**Comment by prav02** [comment]
When I am try to running the code I got this big error message. 
I have crosscheck my connections and credentials, they all are correct.

 UnhandledPromiseRejectionWarning: Error: end: No SFTP connection available
    at formatError (F:\Projects\erp_node\node_modules\ssh2-sftp-client\src\utils.js:62:18)
    at Object.haveConnection (F:\Projects\erp_node\node_modules\ssh2-sftp-client\src\utils.js:612:20)
    at Promise (F:\Projects\erp_node\node_modules\ssh2-sftp-client\src\index.js:1248:19)
    at new Promise (<anonymous>)
    at SftpClient.end (F:\Projects\erp_node\node_modules\ssh2-sftp-client\src\index.js:1236:12)
    at client.connect.then.then.catch.then (F:\Projects\erp_node\node_modules\multer-sftp\lib\index.js:53:33)
    at process._tickCallback (internal/process/next_tick.js:68:7)
(node:22356) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 2)
(node:22356) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.

**Comment by chunkai1312** [comment]
@prav02 This error occurred from [ssh2-sftp-client](https://github.com/theophilusx/ssh2-sftp-client). Please check its documentation for details.

**Comment by prav02** [comment]
Could not find a declaration file for module 'multer-sftp'. 'h:/Work Projects/ERP/erp_node/node_modules/multer-sftp/lib/index.js' implicitly has an 'any' type.
  Try `npm install @types/multer-sftp` if it exists or add a new declaration (.d.ts) file containing `declare module 'multer-sftp';`

I also tried to follow the instructions but no luck
How to solve this error??

**Comment by chunkai1312** [comment]
I've just released TypeScript support in `v1.1.0`.
You can re-install the module and try it again.
Thanks.

**Comment by khasim-osv** [comment]
How did we resolved this issue?

UnhandledPromiseRejectionWarning: Error: end: No SFTP connection available
